### PR TITLE
refactor(ui5-upload-collection-item): migrate to TypeScript

### PR DESCRIPTION
**Pull request by gmkv**
Part of #4337

### CODE REVIEW

File: packages/fiori/src/UploadCollectionItem.ts [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+		if (isSpace(e)) {
+			this._onTerminate();
+		}
+	}
+
+	getFocusDomRef() {
+		return this.getDomRef();
+	}
+
+	get list() {
```
Comment (nnaydenow) [inline packages/fiori/src/UploadCollectionItem.ts:393]: do we really need that getter? I wasn't able to find its usage.

**Review by nnaydenow — COMMENTED** [review]
It looks good to me. @ilhan007 could you double check?

File: packages/fiori/src/UploadCollectionItem.ts (hunk 1 above)
Comment (gmkv) [reply packages/fiori/src/UploadCollectionItem.ts:393]: Looks like old code. Removed it.

File: packages/main/src/ListItem.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import property from "@ui5/webcomponents-base/dist/decorators/property.js";
 import event from "@ui5/webcomponents-base/dist/decorators/event.js";
 import languageAware from "@ui5/webcomponents-base/dist/decorators/languageAware.js";
 import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import UI5Element from "@ui5/webcomponents-base/src/UI5Element.js";
```
Comment (ilhan007) [inline packages/main/src/ListItem.ts:12]: import type UI5Element, you are not using it as a value

**Review by ilhan007 — DISMISSED** [review]

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**